app/models/cloud_service.rb

```diff
@@ -0,0 +1,37 @@
+class CloudService < ApplicationRecord
+  belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "ManageIQ::Providers::CloudManager"
+  belongs_to :host
+  belongs_to :system_service
+  belongs_to :availability_zone
+
+  def fog_service
+    connection_options = {:service => source}
+    ext_management_system.with_provider_connection(connection_options) do |service|
+      service.services.find { |s| s.id.to_s == ems_ref }
+    end
+  end
+
+  def enable_scheduling
+    fog_service.enable
+  end
+
+  def disable_scheduling
+    fog_service.disable
+  end
+
+  def validate_enable_scheduling
+    fog_service && scheduling_disabled?
+  end
+
+  def validate_disable_scheduling
+    fog_service && scheduling_enabled?
+  end
+
+  def scheduling_disabled?
```

Comment:
http://makandracards.com/makandra/607-automagical-methods-for-boolean-attributes-in-activerecord

Seems so